src/linter/ui5Types/bestPractices.ts

```diff
@@ -0,0 +1,358 @@
+			properties[key] = {value: false, node: prop.initializer};
+		} else if (prop.initializer.kind === ts.SyntaxKind.NullKeyword) {
+			properties[key] = {value: null, node: prop.initializer};
+		} else if (ts.isObjectLiteralExpression(prop.initializer) && prop.initializer.properties) {
+			properties[key] = {value: extractPropsRecursive(prop.initializer), node: prop.initializer};
+		} else if (ts.isArrayLiteralExpression(prop.initializer)) {
+			const resolvedValue = prop.initializer.elements.map((elem) => {
+				if (!ts.isObjectLiteralExpression(elem)) {
+					return;
+				}
+				return extractPropsRecursive(elem);
+			}).filter(($) => $) as propsRecordValueType[];
+
+			properties[key] = {value: resolvedValue, node: prop.initializer};
+		} else if (
+			(ts.isIdentifier(prop.initializer) ||
+			ts.isNumericLiteral(prop.initializer) ||
+			ts.isStringLiteral(prop.initializer)) &&
+
+			prop.initializer.getText()) {
+			properties[key] = {value: prop.initializer.getText(), node: prop.initializer};
+		} else {
+			// throw new Error("Unhandled property assignment");
+		}
+	});
+	return properties;
+}
+
+function reportResults(
+	analysisResult: AsyncInterfaceFindType,
+	reporter: SourceFileReporter,
+	classDesc: ts.ClassDeclaration
+) {
+	const {hasAsyncInterface, routingAsyncFlag, rootViewAsyncFlag, hasManifestDefinition} = analysisResult;
+
+	if (!hasManifestDefinition) {
+		reporter.addMessage({
+			node: classDesc,
+			severity: LintMessageSeverity.Warning,
+			ruleId: "ui5-linter-add-manifest",
+			message: "Include a manifest section into the Component.js",
+			messageDetails: "manifest.json will be loaded and used, " +
+			"but is not defined in Component's metadata section",
+		});
+	}
+
+	if (hasAsyncInterface !== AsyncInterfaceStatus.true) {
+		if ([AsyncInterfaceStatus.propNotSet, AsyncInterfaceStatus.false].includes(rootViewAsyncFlag) ||
+			[AsyncInterfaceStatus.propNotSet, AsyncInterfaceStatus.false].includes(routingAsyncFlag)) {
+			const componentsToAddress = [];
+			if (AsyncInterfaceStatus.parentPropNotSet !== rootViewAsyncFlag) {
+				componentsToAddress.push("Root View");
+			}
+			if (AsyncInterfaceStatus.parentPropNotSet !== routingAsyncFlag) {
+				componentsToAddress.push("Routing");
+			}
+
+			reporter.addMessage({
+				node: classDesc,
+				severity: LintMessageSeverity.Error,
+				ruleId: "ui5-linter-no-sync-loading",
+				message: `${componentsToAddress.join(" and ")} ${componentsToAddress.length > 1 ? "are" : "is"} ` +
+				"not configured to load targets asynchronously",
+				messageDetails: "{@link topic:676b636446c94eada183b1218a824717 Use Asynchronous Loading}. " +
+				"Implement sap.ui.core.IAsyncContentCreation interface in Component.js or set async flags for " +
+				"\"sap.ui5/routing/config\" and \"sap.ui5/rootView\" in the manifest.json",
+			});
+		}
+	} else {
+		if (rootViewAsyncFlag === AsyncInterfaceStatus.true) {
+			reporter.addMessage({
+				node: classDesc,
```

Comment:
Wouldn't it be a lot more intuitive for developers to see those messages within the `manifest.json` at the exact line where the flag should be removed? Right now all messages are reported in the Component.js which can be confusing. I think that's fine for the other messages, as the preferred solution is to add the interface in Component.js, but in this case a change should only be made in manifest.json

Reply:
Done with https://github.com/SAP/ui5-linter/pull/73/commits/b3c1bb53805268957f238249fbb97a74b74f25da